src/wms/src/main/java/org/geoserver/wms/map/GeoTIFFMapResponse.java

```diff
@@ -81,8 +87,21 @@ public void formatImageOutputStream(RenderedImage image, OutputStream outStream,
         image = applyPalette(image, mapContent, IMAGE_GEOTIFF8, false);
         
         // crating a grid coverage
-        final GridCoverage2D gc = factory.create("geotiff", image,
+        GridCoverage2D gc = factory.create("geotiff", image,
                 new GeneralEnvelope(mapContent.getRenderingArea()));
+        
+        // NoData stuff
+        if(image instanceof PlanarImage){
```

Comment:
This information is added only in case NoData has been defined as a property for the input image, otherwise everything will be left untouched.